rs/bitcoin/kyt/src/dashboard.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{state, BtcNetwork};
+use askama::Template;
+use ic_btc_interface::Txid;
+use state::{FetchTxStatus, Timestamp};
+
+#[cfg(test)]
+mod tests;
+
+mod filters {
+    pub fn timestamp_to_datetime<T: std::fmt::Display>(timestamp: T) -> askama::Result<String> {
+        let input = timestamp.to_string();
+        let ts: i128 = input
+            .parse()
+            .map_err(|e| askama::Error::Custom(Box::new(e)))?;
+        let dt_offset = time::OffsetDateTime::from_unix_timestamp_nanos(ts).unwrap();
+        // 2020-12-09T17:25:40+00:00
+        let format =
+            time::format_description::parse("[year]-[month]-[day]T[hour]:[minute]:[second]+00:00")
+                .unwrap();
+        Ok(dt_offset.format(&format).unwrap())
+    }
+}
+
+#[derive(Template)]
+#[template(path = "dashboard.html")]
+pub struct DashboardTemplate {
+    btc_network: BtcNetwork,
+    outcall_capacity: u32,
+    fetch_tx_status: Vec<(Txid, Timestamp, Status)>,
+}
+
+#[derive(Debug, Clone)]
+pub enum Status {
+    PendingOutcall,
+    PendingRetry,
+    Error(String),
+    Fetched(Fetched),
+}
+
+#[derive(Debug, Clone)]
+pub struct Fetched {
+    input_addresses: Vec<Option<bitcoin::Address>>,
+}
+
+impl Status {
+    pub fn to_str(&self) -> &str {
+        match self {
+            Status::PendingOutcall => "Pending outcall",
+            Status::PendingRetry => "Pending retry",
+            Status::Error(_) => "Error",
+            Status::Fetched(_) => "Fetched",
+        }
+    }
+    pub fn is_fetched(&self) -> bool {
+        matches!(self, Status::Fetched(_))
+    }
+    pub fn as_fetched(&self) -> &Fetched {
```

Comment:
nit: I would not expect an `as_` method to panic (see [C-CONV](https://rust-lang.github.io/api-guidelines/naming.html#c-conv)). We could rename it, but I think you don't exactly need for the template. E.g, you could do the following

```rust
pub fn fetched(&self) -> Option<&Fetched> {
    match self {
        Status::Fetched(fetched) => Some(fetched),
        _ => None,
    }
}
```

Then in the template `dashboard.html` you can use the [If Let syntax](https://djc.github.io/askama/template_syntax.html#if-let)
```
{% if let Some(fetched) = status.fetched() %}
<tbody>
    {% for address in fetched.input_addresses %}
    <tr><td>{% call btc_address(address) %}</td></tr>
    {% endfor %}
</tbody>
{% endif %}
```
Which also removes the need for `Status::is_fetched`. Similar comment for `is_error/as_error`.

Reply:
Thanks for the suggestion! Fixed.